crates/sui-core/src/authority.rs

```diff
@@ -2062,6 +2062,48 @@ impl AuthorityState {
         }
     }
 
+    pub async fn multi_get_transactions(
+        &self,
+        digests: &[TransactionDigest],
+    ) -> Result<VerifiedTransactionBatch, anyhow::Error> {
+        let transactions = self.database.multi_get_transactions(digests)?;
+        let effects = self.database.multi_get_executed_effects(digests)?;
+        let checkpoints = self.database.multi_get_transaction_checkpoint(digests)?;
+        let mut response: VerifiedTransactionBatch = Vec::new();
+
+        if transactions.len() == effects.len() && checkpoints.len() == transactions.len() {
+            for ((transaction, effect), checkpoint) in transactions
+                .iter()
+                .zip(effects.iter())
+                .zip(checkpoints.iter())
+            {
+                match ((transaction.clone(), effect.clone()), &checkpoint.clone()) {
+                    ((Some(transaction), Some(effect)), check) => {
+                        response.push(((transaction, effect), *check))
+                    }
+                    ((_, _), _) => continue,
+                }
+            }
+        }
+
+        if response.len() != digests.len() {
+            let mut missed_digests: Vec<TransactionDigest> = Vec::new();
+            for digest in digests.iter() {
+                if effects
+                    .iter()
+                    .any(|effect| effect.clone().unwrap().transaction_digest != *digest)
+                {
+                    missed_digests.push(*digest);
+                }
+            }
+            Err(anyhow!(SuiError::TransactionsNotFound {
+                digests: missed_digests
+            }))
+        } else {
+            Ok(response)
+        }
+    }
```

Comment:
if you can ensure the order of multi_get_transaction and multi_get_transaction_checkpoint (we can ensure it by doing [this](https://github.com/MystenLabs/sui/blob/513c2902a1af9aa62445f9aa62c8aeabf8fe2cc1/crates/sui-core/src/authority/authority_store.rs#L193-L196)), you can zip all of them together and check for missing data: 
```
        let mut missed_digests = vec![];
        let mut response: VerifiedTransactionBatch = vec![];

        for data in izip!(digests, transactions, effects, checkpoints) {
            match data {
                (_, Some(tx), Some(effect), cp) => response.push(((tx, effect), cp)),
                (digest, _, _, _) => missed_digests.push(*digest),
            }
        }

        if !missed_digests.is_empty() {
            Err(anyhow!(SuiError::TransactionsNotFound {
                digests: missed_digests
            }))
        } else {
            Ok(response)
        }
```